ibm/service/cos/resource_ibm_cos_backup_policy.go

```diff
@@ -0,0 +1,153 @@
+	backupType := d.Get("backup_type").(string)
+	rcClient, err := meta.(conns.ClientSession).CosConfigV1API()
+	if err != nil {
+		return diag.Errorf("Failed to create rc client %v", err)
+	}
+	createBucketBackupPolicyOptions := &rc.CreateBackupPolicyOptions{
+		Bucket:               aws.String(bucketName),
+		PolicyName:           aws.String(policyName),
+		TargetBackupVaultCrn: aws.String(targetBackupVaultCRN),
+		BackupType:           aws.String(backupType),
+	}
+	res, _, err := rcClient.CreateBackupPolicy(createBucketBackupPolicyOptions)
+	if err != nil {
+		return diag.Errorf("Failed to create the backup policy %s, %v", bucketName, err)
+	}
+	policyId := *res.PolicyID
+	policySetID := fmt.Sprintf("%s:%s:%s:target:%s", bucketName, policyId, policyName, targetBackupVaultCRN)
+	d.SetId(policySetID)
+	return resourceIBMCOSBackupPolicyRead(ctx, d, meta)
+}
+
+func resourceIBMCOSBackupPolicyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketName := parseBackupPolicyID(d.Id(), "bucketName")
+	policyId := parseBackupPolicyID(d.Id(), "policyId")
+	rcClient, err := meta.(conns.ClientSession).CosConfigV1API()
+	if err != nil {
+		return diag.Errorf("Failed to create rc client %v", err)
+	}
+	d.Set("policy_id", policyId)
+	getBucketBackupPolicyOptions := &rc.GetBackupPolicyOptions{
+		Bucket:   aws.String(bucketName),
+		PolicyID: aws.String(policyId),
+	}
+
+	res, _, err := rcClient.GetBackupPolicy(getBucketBackupPolicyOptions)
+	if err != nil {
+		return diag.Errorf("Error while reading the backup policy vault : %v", err)
+	}
+	if res != nil {
+		if res.PolicyName != nil {
+			d.Set("policy_name", aws.String(*res.PolicyName))
+		}
+		if res.TargetBackupVaultCrn != nil {
+			d.Set("target_backup_vault_crn", aws.String(*res.TargetBackupVaultCrn))
+		}
+		if res.BackupType != nil {
+			d.Set("backup_type", aws.String(*res.BackupType))
+		}
+	}
+	return nil
+}
+
+func resourceIBMCOSBackupPolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketName := parseBackupPolicyID(d.Id(), "bucketName")
+	policyId := parseBackupPolicyID(d.Id(), "policyId")
+	rcClient, err := meta.(conns.ClientSession).CosConfigV1API()
+	if err != nil {
+		return diag.Errorf("Failed to create rc client %v", err)
+	}
+	deleteBucketBackupPolicyOptions := &rc.DeleteBackupPolicyOptions{
+		Bucket:   aws.String(bucketName),
+		PolicyID: aws.String(policyId),
+	}
+	_, deleteErr := rcClient.DeleteBackupPolicy(deleteBucketBackupPolicyOptions)
+	if deleteErr != nil {
+		return diag.Errorf("Error deleting the backup policy %v", deleteErr)
+	}
+	return nil
+}
+
+func parseBackupPolicyID(id string, info string) string {
+	if info == "bucketName" {
+		return strings.Split(id, ":")[0]
+	}
+	if info == "policyId" {
+		return strings.Split(id, ":")[1]
+	}
+	if info == "policyName" {
+		return strings.Split(id, ":")[2]
+	}
+	if info == "targebucketCrn" {
+		return strings.Split(id, ":target:")[1]
+	}
+	return ""
```

Comment:
Handle to return proper error when some one provide invalid id for import scenario

Reply:
Done